### DISCUSSION TITLE
Props are undefined from getServerSideProps

### DISCUSSION THREAD

**Discussion by colton-savage-lmp** [vercel/next.js#14508] [Help] [answered]
# Bug report

## Describe the bug

As described in the [documentation](https://nextjs.org/docs/basic-features/data-fetching#typescript-use-getserversideprops), the results returned from `getServersideProps` is undefined.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

With a basic Next Typescript project, create a new page with the following:

```
import React from 'react';
import { GetServerSideProps, InferGetServerSidePropsType } from 'next';

export const getServerSideProps: GetServerSideProps = async () => {
	return {
		props: {
			results: [1,2,3],
		},
	};
};

function Page({results}: InferGetServerSidePropsType<typeof getServerSideProps>) {
	console.log(results) // undefined
	return (
		<div className="app__page header">page here</div>
	)
}

export default Page;
```

## Expected behavior

The value of the console.log should be `[1,2,3]`

## System information

- OS: macOS
- Version of Next.js: 9.4.4
- Version of Node.js: 12.18.0

**Comment by Timer** [comment]
Can you share your `pages/_app.js` file?

**Comment by colton-savage-lmp** [comment, marked as answer]
@Timer I can if you can excuse the current messiness 😅 

`_app.tsx`

```
import React, { createContext, useEffect, useState } from 'react';
import Head from 'next/head';

import Navbar from '../components/Navbar';

import '../app.scss'

interface AppProps {
	Component: React.FC,
	props: any,
};

interface Customer {
	name: string,
	id: string,
	email: string,
	postalCode: string,
};

interface AppContextInterface {
	amazonSDK: any,
	setAmazon: (value: any) => void,
	setCustomer: (value: any) => void,
	customer: Customer | undefined,
};

export const AppContext = createContext<AppContextInterface>({
	amazonSDK: undefined,
	customer: undefined,
	setAmazon: () => {},
	setCustomer: () => {},
});

declare global {
	interface Window { onAmazonLoginReady: any; }
};

declare const amazon: any;


const App: React.FC<AppProps> = ({ Component, props }) => {
	const [amazonSDK, setAmazon] = useState();
	const [customer, setCustomer] = useState();
	const amazonValue = { amazonSDK, setAmazon, customer, setCustomer };
	const [isComponentMounted, setIsComponentMounted] = useState(false)

	useEffect(() => setIsComponentMounted(true), [])

	useEffect(() => {
		window.onAmazonLoginReady = function() {
			setAmazon(amazon);
			amazon.Login.setClientId('');
		};
	}, [])

	return (
		<AppContext.Provider value={amazonValue}>
			<Head>
				<meta charSet="utf-8" />
				<meta name="viewport" content="width=device-width, initial-scale=1" />
				<meta name="theme-color" content="#000000" />
				<meta
					name="description"
					content="Demo site for LWA and dev environment setup"
				/>
				<title>BLANK</title>
				{isComponentMounted &&
					<script dangerouslySetInnerHTML={{__html: `
						(function(d) {
							var a = d.createElement('script'); a.type = 'text/javascript';
							a.async = true; a.id = 'amazon-login-sdk';
							a.src = 'https://assets.loginwithamazon.com/sdk/na/login1.js';
							d.getElementById('amazon-root').appendChild(a);
						})(document);
					`}}></script>
				}
			</Head>
			<Navbar />
			<Component {...props} />
			<div id="amazon-root"></div>
		</AppContext.Provider>
	)
};

export default App;
```

**Reply by timneutkens** [reply]
https://nextjs.org/docs/advanced-features/custom-app

The signature is `{ Component, pageProps }`, you're using `{ Component, props }` so the page props are not being passed into the component, hence why you're not seeing the getServerSideProps result.

This was caused by the custom `AppProps` type that you have. Next.js includes the correct type: https://nextjs.org/docs/basic-features/typescript#custom-app

**Reply by colton-savage-lmp** [reply]
Ah I see, thank you!

**Comment by j-lee8** [comment]
Getting the same issue. Very strange. I've added getServerSideProps into my `CarTab.tsx` inside page (`car/[id].tsx`) and it's always undefined. I've even created an index.tsx inside car/ and it's still undefined. Is this an issue with Next or am I missing something really obvious? Thanks all

This is my _app.tsx:

```
<Provider store={store}>
	<PersistGate persistor={persistor}>
		<NhostAuthProvider auth={auth}>
			<NhostApolloProvider
				auth={auth}
				gqlEndpoint="..."
			>
				<ChakraProvider theme={customTheme} resetCSS>
					<DialogProvider>
						<BaseModal />
						<Component {...pageProps} />
					</DialogProvider>
				</ChakraProvider>
			</NhostApolloProvider>
		</NhostAuthProvider>
	</PersistGate>
</Provider
```
---------------------------------------------------------------------------
`car/[id].tsx (<CarTab />)`

```
function CarTab({ data }) {
	console.log('idea: ', data);
        return "test";
}

export async function getServerSideProps() {
	const data = { title: 'test car' };
	return { props: data };
}
```

**Comment by Shreyas-29** [comment]
i am fetching data from the json api in index.tsx
like this

`export const getServerSideProps: GetServerSideProps = async (context: GetServerSidePropsContext) => {

  const res = await fetch('https://mocki.io/v1/3bf76aa7-e669-4289-900d-4153eb840fd2');
  const data = await res.json();

  const session = await getSession();

  const providers = await getProviders();

  return {
    props: {
      session,
      providers,
      data
    },
  };
};
;`

and getting the data as undefined!!

`export default function Home({ providers, data }: Props) {

  const { data: session } = useSession();
  console.log('data>>>', data)

  if (!session) return <LoginPage providers={providers} />;

  return (
    <>
      <Head>
        <title>Facebook - Home</title>
        <meta name="description" content="Facebook clone by Shreyas" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/logo.png" />
      </Head>
      <Header />
      <main className='!bg-[#f0f2f5] flex'>
        <Sidebar />
        <Feed />
        <Widgets />
      </main>
    </>
  )
}`

all the props session, providers and data getting undefined
Please anyone help me to solve this issue

**Comment by AirmanEpic** [comment]
Still getting this bug in 14.2.3
